src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,390 @@
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/matcher"
+
+module CFA
+  # class representings /etc/ntp.conf file model. It provides helper to manipulate
+  # with file. It uses CFA framework and Augeas parser.
+  # @see http://www.rubydoc.info/github/config-files-api/config_files_api/CFA/BaseModel
+  # @see http://www.rubydoc.info/github/config-files-api/config_files_api/CFA/AugeasParser
+  #
+  # In NTP files some keys can be present multiple times.
+  #
+  # For example:
+  #
+  # server 1.pool.ntp.org iburst
+  # server 2.pool.ntp.org
+  # server 127.127.1.0
+  # fudge 127.127.1.0 stratum 10
+  # peer 128.100.0.45
+  # peer 192.168.1.30
+  #
+  # Order of lines is not important, except for the
+  # fudge option that should immediately follow
+  # appropriate "server" option.
+  #
+  # Important keys (that are to be read and written) are
+  # server, peer, fudge, broadcast and broadcastclient.
+  class NtpConf < ::CFA::BaseModel
+    PARSER = CFA::AugeasParser.new("ntp.lns")
+
+    PATH = "/etc/ntp.conf".freeze
+
+    RECORD_ENTRIES = %w(
+      server
+      peer
+      broadcast
+      broadcastclient
+      manycast
+      manycastclient
+      fudge
+      restrict
+    ).freeze
+
+    COLLECTION_KEYS = (RECORD_ENTRIES + ["action"]).freeze
+
+    def initialize(file_handler: nil)
+      super(PARSER, PATH, file_handler: file_handler)
+    end
+
+    # Loads and fixes AugeasElement keys.
+    #
+    # AugeasParser returns collection entries with
+    # key ending in '[]', for example 'server[]'.
+    #
+    # If only exists one entry of the collection, parser
+    # return key without '[]'.
+    #
+    # For example:
+    #
+    # server 1.pool.ntp.org iburst
+    # server 2.pool.ntp.org
+    # peer 128.100.0.45
+    #
+    # In this case, key for peer entry is 'peer'.
+    # It is necessary to add '[]' to avoid possible
+    # write issues when more entries of that type
+    # are added.
+    def load
+      super
+      fix_keys(data)
+    end
+
+    # Obtains a collection that represents the
+    # entries of the file.
+    # @return [CollectionRecord] collection to
+    #   manipulate ntp entries.
+    #
+    # The collection preserves the order of the
+    # ntp entries in the file and only contains the
+    # entries of interest (see RECORD_ENTRIES).
+    def records
+      RecordCollection.new(data)
+    end
+
+    # Obtains raw content of the file.
+    # @return [String]
+    def raw
+      PARSER.serialize(data)
+    end
+
+  private
+
+    def fix_keys(tree)
+      tree.data.each do |entry|
+        entry[:key] << "[]" if COLLECTION_KEYS.include?(entry[:key])
+        fix_keys(entry[:value].tree) if entry[:value].is_a?(AugeasTreeValue)
+      end
+    end
+
+    # class to manage ntp entries as a collection.
+    #
+    # Each element of the collection is a subclass of Record.
+    # A Record object is a wrapper of an AugeasElement.
+    class RecordCollection
```

Comment:
I just notice one thing. this class is in private section, but in fact it is public API as above method records return it, so it should be also properly documented, especially public API as user of `records` method need to know it.